source/fem-dem/void_fraction.cc

```diff
@@ -0,0 +1,1141 @@
+                      const double r_particle =
+                        particle_properties[DEM::PropertiesIndex::dp] * 0.5;
+                      double single_particle_volume =
+                        M_PI * Utilities::fixed_power<dim>(r_particle * 2.0) /
+                        (2 * dim);
+
+                      // Distance between particle and quadrature point
+                      // centers
+                      distance = particle.get_location().distance(
+                        quadrature_point_location[q]);
+
+                      // Particle completely in reference sphere
+                      if (distance <= (r_sphere - r_particle))
+                        particles_volume_in_sphere +=
+                          (M_PI *
+                           Utilities::fixed_power<dim>(
+                             particle_properties[DEM::PropertiesIndex::dp]) /
+                           (2.0 * dim)) *
+                          single_particle_volume /
+                          particle_properties
+                            [DEM::PropertiesIndex::volumetric_contribution];
+
+                      // Particle completely outside reference sphere. Do
+                      // absolutely nothing.
+
+                      // Particle partially in reference sphere
+                      else if ((distance > (r_sphere - r_particle)) &&
+                               (distance < (r_sphere + r_particle)))
+                        {
+                          if (dim == 2)
+                            particles_volume_in_sphere +=
+                              particle_circle_intersection_2d(r_particle,
+                                                              r_sphere,
+                                                              distance) *
+                              single_particle_volume /
+                              particle_properties
+                                [DEM::PropertiesIndex::volumetric_contribution];
+                          else if (dim == 3)
+                            particles_volume_in_sphere +=
+                              particle_sphere_intersection_3d(r_particle,
+                                                              r_sphere,
+                                                              distance) *
+                              single_particle_volume /
+                              particle_properties
+                                [DEM::PropertiesIndex::volumetric_contribution];
+                        }
+                    }
+                }
+
+              // Execute same operations for periodic neighbors, if the
+              // simulation has no periodic boundaries, the container is
+              // empty. Also, those operations can not be done in the previous
+              // loop because the particles on the periodic side needs a
+              // correction with an offset for the distance with the
+              // quadrature point
```

Comment:
```suggestion
              // Execute same operations for periodic neighbors, if the
              // simulation has no periodic boundaries, the container is
              // empty. Also, those operations cannot be done in the previous
              // loop because the particles on the periodic side need a
              // correction with an offset for the distance with the
              // quadrature point
```